qhub/initialize.py

```diff
@@ -255,7 +259,10 @@ def render_config(
     config["provider"] = cloud_provider
 
     if ci_provider is not None and ci_provider != "none":
-        config["ci_cd"] = {"type": ci_provider, "branch": "main"}
+        config["ci_cd"] = CICD_CONFIGURATION.copy()
+        config["ci_cd"]["type"] = ci_provider
+        if os.environ.get("QHUB__cicd__commit_render") == "False":
```

Comment:
No need for this `if` statement. Have plans for a format complete implementation for setting arbitrary values via environment variables.